File: examples/bubble_sort.ts
/**
 * Bubble sort implementation in TypeScript
 */
function bubbleSort(arr: number[]): number[] {
    const n = arr.length;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                // Swap elements
                [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
            }
        }
    }
    return arr;
}

// Example usage
const numbers: number[] = [64, 34, 25, 12, 22, 11, 90];
console.log("Original array:", numbers);
const sortedNumbers = bubbleSort([...numbers]);
console.log("Sorted array:", sortedNumbers);

export { bubbleSort };

File: examples/bubble_sort.py
def bubble_sort(arr):
    """
    Bubble sort implementation in Python
    """
    n = len(arr)
    for i in range(n):
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
    return arr


if __name__ == "__main__":
    numbers = [64, 34, 25, 12, 22, 11, 90]
    print("Original array:", numbers)
    sorted_numbers = bubble_sort(numbers.copy())
    print("Sorted array:", sorted_numbers)
